File: python/7 kyu/294.py
# https://www.codewars.com/kata/667dfbaa4570b2db26aedc8c/train/python

# Passed

def size_to_number(size):
    if len(size) == 0:
        return None
    
    base = size[-1]
    base_to_size_dict = {
                         "s": 36, 
                         "m": 38, 
                         "l": 40
                        }
    if base not in base_to_size_dict:
        return None
    
    if base == "m" and len(size) > 1:
        return None
    
    size_value = base_to_size_dict[base]
    if len(size) > 1:
        for modifier in size[:-1]:
            if modifier != "x":
                return None
            
            if base == "l":
                size_value += 2
            elif base == "s":
                size_value -= 2
    
    return size_value

output = size_to_number("xxxs")
print(output)

File: nodejs/7 kyu/294.js
// https://www.codewars.com/kata/5ef0456fcd067000321baffa/train/javascript

// Passed

function connotation(str) {
  let words = str.split(" ").filter((word) => word.length > 0);
  let wordsSide = words.map((word) =>
    word.toLowerCase()[0].charCodeAt() <= 109 ? 1 : -1,
  ); // arr of 1 or -1
  let overallSide = wordsSide.reduce((a, b) => a + b, 0);
  let result = overallSide >= 0;
  return result;
}

const output = connotation("CHOCOLATE MAKES A GREAT SNACK");
console.log(output);